docs: Document Preference Center slot label customization (#542)
mautic/developer-documentation-new: docs/plugin_extensions/landing_pages.rst

            $event->setContent($content);
        }
    }

.. vale off

Customizing Preference Center Pages
***********************************

.. vale on

> Preference Center pages let Contacts manage their communication preferences. In Mautic 7.1 and later, you can programmatically customize the labels on Preference Center slot components using the ``PageEvents::PAGE_ON_DISPLAY`` event.

Author: Added Preference Center slot label customization documentation based on PR #16041 which introduced customizable `label-text` attributes for `segmentlist`, `categorylist`, `preferredchannel`, `channelfrequency`, and `saveprefsbutton` slots. The PR description confirms this customization can only be done programmatically via the `PageEvents::PAGE_ON_DISPLAY` event.

Source: https://github.com/mautic/mautic/pull/16041

.. note::

   The ``TokenHelper::REGEX`` constant provides the regular expression pattern for matching Contact field tokens. Use this to check whether a URL contains tokens before performing replacements.

.. vale off

Customizing Preference Center Pages

> ***********************************

Reviewer: ```suggestion
Customizing Preference Center
*****************************
```

.. vale on

> Preference Center pages let Contacts manage their communication preferences. In Mautic 7.1 and later, you can programmatically customize the labels on Preference Center slot components using the ``PageEvents::PAGE_ON_DISPLAY`` event.

Reviewer: ```suggestion
.. vale off

Preference Center lets Contacts manage their communication preferences. In Mautic 7.1 and later, you can programmatically customize the labels on Preference Center slot components using the ``PageEvents::PAGE_ON_DISPLAY`` event.

.. vale on
```

This lets you override default translated labels with custom text for branding, localization beyond built-in translations, or dynamic label generation based on context.

Available slot parameters
=========================

Each Preference Center slot accepts label attributes that override the default translated text:

.. vale off

.. list-table::
   :header-rows: 1
   :widths: 25 25 50

   * - Slot
     - Attribute
     - Default translation key
   * - ``segmentlist``
     - ``label-text``

>      - ``mautic.lead.form.list`` ('My segments')

Reviewer: ```suggestion
     - ``mautic.lead.form.list`` - **My segments**
```

   * - ``categorylist``
     - ``label-text``

>      - ``mautic.lead.form.categories`` ('My categories')

Reviewer: ```suggestion
     - ``mautic.lead.form.categories`` - **My categories**
```

   * - ``preferredchannel``
     - ``label-text``

>      - ``mautic.lead.list.frequency.preferred.channel`` ('I prefer communication by')

Reviewer: ```suggestion
     - ``mautic.lead.list.frequency.preferred.channel`` - **I prefer communication by**
```

   * - ``channelfrequency``
     - ``label-text``

>      - ``mautic.lead.contact.me.label`` ('I want to receive %channel%')

Reviewer: ```suggestion
     - ``mautic.lead.contact.me.label`` - **I want to receive %channel%**
```

   * - ``channelfrequency``
     - ``label-text1``

>      - ``mautic.lead.list.frequency.number`` ('Do not contact more than')

Reviewer: ```suggestion
     - ``mautic.lead.list.frequency.number`` - **Do not contact more than**
```

   * - ``channelfrequency``
     - ``label-text2``

>      - ``mautic.lead.list.frequency.times`` ('Messages each')

Reviewer: ```suggestion
     - ``mautic.lead.list.frequency.times`` - **Messages each**
```

   * - ``channelfrequency``
     - ``label-text3``

>      - ``mautic.lead.frequency.dates.label`` ('Pause from')

Reviewer: ```suggestion
     - ``mautic.lead.frequency.dates.label`` - **Pause from**
```

   * - ``channelfrequency``
     - ``label-text4``

>      - ``mautic.lead.frequency.contact.end.date`` ('Pause to')

Reviewer: ```suggestion
     - ``mautic.lead.frequency.contact.end.date`` **Pause to**
```

   * - ``saveprefsbutton``
     - ``btnText``

>      - ``mautic.page.form.saveprefs`` ('Save preferences')

Reviewer: ```suggestion
     - ``mautic.page.form.saveprefs`` - **Save preferences**
```

.. vale off

Customizing Preference Center
*****************************

> Preference Center lets Contacts manage their communication preferences. In Mautic 7.1 and later, you can programmatically customize the labels on Preference Center slot components using the ``PageEvents::PAGE_ON_DISPLAY`` event.

Reviewer: As the PR at `mautic/mautic` pointing to 7.x, is this feature available since Mautic 7.2?
